include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -728,13 +733,15 @@ namespace internal
               MPI_STATUS_IGNORE);
             AssertThrowMPI(ierr_3);
 
-            for (unsigned int i = 0; i < buffer.size();
-                 i += dof_handler_fine.get_fe(buffer[i]).n_dofs_per_cell() + 1)
-              ghost_indices.insert(
-                ghost_indices.end(),
-                buffer.begin() + i + 1,
-                buffer.begin() + i + 1 +
-                  dof_handler_fine.get_fe(buffer[i]).n_dofs_per_cell());
+            for (unsigned int i = 0; i < buffer.size();)
+              {
+                const unsigned int dofs_per_cell =
+                  dof_handler_fine.get_fe(buffer[i]).n_dofs_per_cell();
+                ++i;
+                for (unsigned int j = 0; j < dofs_per_cell; ++j, ++i)
+                  if (!is_extended_locally_owned.is_element(buffer[i]))
+                    ghost_indices.push_back(buffer[i]);
```

Comment:
```suggestion
                  {
                    AssertIndexRange(i, buffer.size());
                    if (!is_extended_locally_owned.is_element(buffer[i]))
                      ghost_indices.push_back(buffer[i]);
                  }
```
Since we are incrementing `i` repeatedly I'd prefer to be safe and have an extra check here.